src/borg/archiver.py

```diff
@@ -2284,6 +2362,18 @@ def run(self, args):
             logger.warning("Using a pure-python msgpack! This will result in lower performance.")
         return args.func(args)
 
+    def _get_archives_slice(self, args, manifest):
+        if args.location.archive:
+            logger.error('The options --oldest and --latest must not be used on archive targets.')
+            self.exit_code = EXIT_ERROR
+            return []
+        n = args.oldest or args.latest
```

Comment:
maybe add a "assert n > 0" here - this code should be only called if either oldest or latest is given and > 0.